K/KaHyPar/build_tarballs.jl

```diff
@@ -0,0 +1,43 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "KaHyPar"
+version = v"1.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/arbenson/kahypar.git", "d64498068ab745a09a94391e746d35853fc3eef3"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd kahypar/
+git submodule update --init --recursive
+mkdir build
+cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_BUILD_TYPE=Release
```

Comment:
The [recommended options](https://github.com/JuliaPackaging/Yggdrasil/wiki/Tips-for-Building-Packages#cmake-builds) for CMake are
```suggestion
cmake .. -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release
```
You're missing ` -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}` which is quite important here

Reply:
This actually breaks the build, which is why I omitted it. Any ideas on why that would happen?

Reply:
More specifically, at some point the build tries to execute a command
`cd /workspace/srcdir/kahypar/build/tools && mtx_to_hgr_converter_test`
The second part of that command is the program name but a path isn't being specified, so it fails. For example, this works:
`cd /workspace/srcdir/kahypar/build/tools && ./mtx_to_hgr_converter_test`
(adding the ./ for the path).

I'm not quite sure how the googletest is interacting with cmake to produce such errors.

Reply:
Well, we do a cross-compilation and that option properly sets the relevant cross-compilation tools, so yes, it's fundamental in our case.

Regarding the attempt to run commands, that's a big problem generally speaking.  You may want to read to understand the problem https://github.com/JuliaPackaging/Yggdrasil/wiki/Build-Tricks#running-foreign-executables

Reply:
I didn't find much in your link, but I think this is going to be difficult to fix within this PR.

Is it worth opening an issue with BinaryBuilder.jl, where the wizard says
"The recommended options for CMake are...
`cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release`
to specify that `-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}` is required?

Reply:
I don't know what's wrong with the tests, but we can't run them anyway, so I simply added a patch to disable all tests.  Well-written build scripts would provide an option to disable tests, instead of having to edit these files.